Assets/Scripts/GameMaster/GameMasterBehavior.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class GameMasterBehavior : NetworkBehaviour
+{
+  public GameObject TetrominoFactory;
+
+  private TetrominoFactoryBehavior _tetrominoFactory;
+  private IList<PlayerBehavior> _players;
+  private IList<TetrominoBehavior> _tetrominos;
+
+  private IDictionary<PlayerBehavior, TetrominoBehavior> _activeTetrominoByPlayer;
+
+  // Use this for initialization
+  public void Start () {
+    _players = new List<PlayerBehavior>();
+    _tetrominos = new List<TetrominoBehavior>();
+    _activeTetrominoByPlayer = new Dictionary<PlayerBehavior, TetrominoBehavior>();
+    _tetrominoFactory = TetrominoFactory.GetComponent<TetrominoFactoryBehavior>();
+  }
+
+  // Update is called once per frame
+  public void Update () {
+    if(FindObjectsOfType<PlayerBehavior>().Count() < 2)
+    {
+      // Do not start the game until we have 2 palyers connected.
+      return;
+    }
+
+    SpawnNewTetrominosIfNeeded();
+  }
+
+  public void RegisterPlayer(PlayerBehavior playerToRegister)
+  {
+    _players.Add(playerToRegister);
+  }
+
+  public void RegisterTetromino(TetrominoBehavior tetrominoToRegister)
+  {
+    _tetrominos.Add(tetrominoToRegister);
+  }
+
+  public void ProcessPlayerInput(PlayerBehavior player, Direction input)
+  {
+    if(!_activeTetrominoByPlayer.ContainsKey(player))
+    {
+      // The player has not been assigned an active tetromino. Do nothing.
+      return;
+    }
+
+    var activeTetromino = _activeTetrominoByPlayer[player];
+    activeTetromino.Move(input);
+  }
+
+  private void SpawnNewTetrominosIfNeeded()
+  {
+    if (_players.All(p => _activeTetrominoByPlayer.ContainsKey(p) && _activeTetrominoByPlayer[p] != null))
+    {
+      // If any players are still controlling an active tetromino, do not spawn new tetrominos.
+      return;
+    }
+
+    foreach(var player in _players)
+    {
+       var tetromino = _tetrominoFactory.SpawnTetromino();
+
+      if (!_activeTetrominoByPlayer.ContainsKey(player))
+      {
+        _activeTetrominoByPlayer.Add(player, tetromino);
+      }
+      else
+      {
+        _activeTetrominoByPlayer[player] = tetromino;

```

Comment:
Why reassign a new object if there's already one for the player?

Reply:
Player is the key. This assigns the new tetromino to the player via the mapping dictionary _activeTetrominoByPlayer. Idea with this method is that if all the players no longer have an active tetromino (where "active" means that the player can control it) we should spawn them new ones. A bit of game logic leaking into this, I guess.

It would normally be more natural for the player to have a direct reference to the tetromino it controls. But that causes problems because PlayerBehavior is a networked object with player issued [Command]s. Using the mapping table on the GameMasterBehavior, which is server-only lets us break from the networky concerns of PlayerBehavior and  keep the tetrominos and GameMasterBehavior from needing to be network aware.